config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
+// protocol version being spoken, some otherwise understood RPC messages may be
+// refused. See isVersionCompatible for details of this logic.
+//
+// There are notes about the upgrade path in the description of the versions
+// below. If you are starting a fresh cluster then there's no reason not to
+// jump right to the latest protocol version. If you need to interoperate with
+// older, unversioned Raft servers you'll need to drive the cluster through the
+// different versions in order.
+//
+// The version details are complicated, but here's a summary of what's required
+// to get from an unversioned cluster to version 2:
+//
+// 1. In version N of your app that starts using the new Raft library with
+//    versioning, set ProtocolVersion to 0.
+// 2. Make version N+1 of your app require version N as a prerequisite (all
+//    servers must be upgraded). For version N+1 of your app set ProtocolVersion
+//    to 1.
+// 3. Similarly, make version N+2 of your app require version N+1 as a
+//    prerequisite. For version N+2 of your app, set ProtocolVersion to 2. Also,
+//    for this upgrade users need to remove each server from the cluster by
+//    address before upgrading that server, and then add it back by ID after the
+//    upgrade.
+//
+// Version History
+//
+// 0: Unversioned, original protocol, used to interoperate with unversioned
```

Comment:
The min version is supposed to be what this server understands, and has to understand version 0 to do a rolling upgrade from 0 to 1. Perhaps the minimum version a server understands ought to be different from the minimum version it may run.